rep-2002.rst

```diff
@@ -0,0 +1,136 @@
+REP: 2002
+Title: ROS 2 Rolling Release
+Author: Steven! Ragnarok <steven@openrobotics.org>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 05-Sep-2019
+Post-History: 05-Sep-2019, 03-Oct-2019, 18-Oct-2019
+
+
+Abstract
+========
+
+This REP describes a rolling ROS 2 distribution which will serve as the target for ongoing development of ROS 2.
+It also describes how future releases of ROS 2 may be produced by taking a snapshot of the rolling distribution to produce the next fixed release of ROS 2. 
+
+
+Motivation
+==========
+
+In order to reduce the workload of Open Robotics staff and individual package maintainers when bootstrapping new ROS releases a rolling ROS distribution will be created to serve as the foundation for subsequent stable ROS distributions.
+
+
+Support status
+==============
+
+The rolling release itself will not be considered a supported platform.
+Distribution syncs will be performed on a regular cadence just like stable distributions but regressions may not block a rolling release sync.
+During the pre-release phase for stable ROS distributions releases in the rolling distribution may be blocked or held back in order to conform to the support requirements of the upcoming stable release.
```

Comment:
I think this is caused by conflating a rolling release and a testing release. This might cause uneven jumps in package versions. Whereas a pure rolling release would have, on average, a steady stream of updates, with this approach you will have a steady stream of updates, then suddenly updates drying up, then one day a massive burst of updates all at once. Then the cycle would be repeated during the next stable release's cycle.

Of course, since the rolling release is not intended for general consumption, this may not be a problem.

Reply:
The release rolls, because we'll be keeping it as evergreen as we can with a shifting platform beneath it, and it serves as a testing ground for forthcoming stable releases.

My intention is to propose a rolling release like Debian `unstable`, rather than like Archlinux. Although unlike Debian unstable we would elide the `testing` release, treating our rolling unstable release as a testing release when a new stable release is coming up. (Which you also already pointed out below).

Do you think that a change in terminology or a further clarification would improve the proposal? What terms would you suggest?

Reply:
I think that if we want to call something testing it would be the state of the release between when it forks off of the rolling release but before it's released as the stable release. (likely the window between January and May, when we have all the target platforms available in prerelease.)

Reply:
That's how Debian does it, effectively. That implies a separate release from the continuously-rolling unstable release. Personally I like that model a lot because it gives developers a chance to see what environment their particular package for an intended release has to sit in, rather than needing to go after a set of moving targets for their dependencies. It also gives the community to try out the release in advance to help test stability without necessariy needing to deal with the bleeding edge.

A pure rolling distribution like Gentoo and Arch do this on a package-by-package basis instead. There are pluses and minuses to that, I think.